modules/consensus/difficulty.go

```diff
@@ -68,10 +68,31 @@ var (
 // however we do not use the child block deltas because that would allow the
 // child block to influence the target of the following block, which makes abuse
 // easier in selfish mining scenarios.
-func (cs *ConsensusSet) childTargetOak(parentTotalTime int64, parentTotalTarget, currentTarget types.Target, parentHeight types.BlockHeight) types.Target {
-	// Determine the detla of the current total time vs. the desired total time.
-	expectedTime := types.BlockFrequency * parentHeight
-	delta := int64(expectedTime) - parentTotalTime
+func (cs *ConsensusSet) childTargetOak(parentTotalTime int64, parentTotalTarget, currentTarget types.Target, parentHeight types.BlockHeight, parentTimestamp types.Timestamp) types.Target {
+	// Determine the delta of the current total time vs. the desired total time.
+	// The desired total time is the difference between the genesis block
+	// timestamp and the current block timestamp.
+	var delta int64
+	if parentHeight < types.OakHardforkFixBlock {
+		// This is the original code. It is incorrect, because it is comparing
+		// 'expectedTime', an absolute value, to 'parentTotalTime', a value
+		// which gets compressed every block. The result is that 'expectedTime'
+		// is substantially larger than 'parentTotalTime' always, and that the
+		// shifter is always reading that blocks have been coming out far too
+		// quickly.
+		expectedTime := int64(types.BlockFrequency * parentHeight)
+		delta = expectedTime - parentTotalTime
+	} else {
+		// This is the correct code. The expected time is an absolute time based
+		// on the genesis block, and the delta is an absolute time based on the
+		// timestamp of the parent block.
+		//
+		// Rules elsewhere in consensus ensure that the timestamp of the parent
+		// block has not been manipulated by more than a few hours, which is
+		// accurate enough for this logic to be safe.
+		expectedTime := int64(types.BlockFrequency*parentHeight) + int64(types.GenesisTimestamp)
+		delta = expectedTime - int64(parentTimestamp)
+	}
```

Comment:
This is the only change of substance in this PR. The first branch of logic in the `if` block should exactly match the original code. Please verify from the diff that this is the case.

The original block of code was incorrectly setting the delta by comparing an absolute time (`expectedTime`) to a time that gets compressed every block (`parentTotalTime`), which always resulted in a massive positive delta, causing the shifter to always target 30 minute blocks.

The new code correctly compares an absolute time (`expectedTime`) to an absoluteTime (`parentTimestamp`).

Reply:
This change does require us to provide the parent timestamp in the function signature, which resulted in us changing `processedblock.go` to correctly use the new function signature.

Reply:
the only thing that could potentially be clearer here is changing the last two lines to:
```go
expectedTime := int64(types.BlockFrequency*parentHeight)
delta = expectedTime - int64(parentTimestamp - types.GenesisTimestamp)
```
that way, the `expectedTime` is the same for both branches. In fact, you could do:
```go
var parentTime int64
if parentHeight < types.OakHardforkFixBlock {
    parentTime = parentTotalTime
} else {
    parentTime = int64(parentTimestamp - types.GenesisTimestamp)
}
delta := expectedTime - parentTime
```
but regardless, the logic seems correct.